src/metabase/lib/equality.cljc

```diff
@@ -412,3 +427,42 @@
          (and (not (contains? matching nil))
               (= (count matching) (count columns))
               (every? #(= (count %) 1) (vals matching))))))
+
+(defn- make-fields-explicit
+  [query field-ids]
+  (if-let [fields (or (->> (lib.util/query-stage query -1)
+                           ;; using lib.field requires circular dep
+                           :fields
+                           (sort-by last)
+                           vec
+                           not-empty)
+                      (when (and (seq field-ids) (:lib/metadata query))
+                        (->> (lib.metadata.calculation/visible-columns query)
```

Comment:
This should probably be `returned-columns`, unless there's something I'm missing?